app/components/user-page/course-progress-list.ts

```diff
@@ -0,0 +1,43 @@
+import Component from '@glimmer/component';
+import { groupBy } from 'codecrafters-frontend/utils/lodash-utils';
+import type UserModel from 'codecrafters-frontend/models/user';
+import type CourseParticipationModel from 'codecrafters-frontend/models/course-participation';
+
+type Signature = {
+  Element: HTMLDivElement;
+
+  Args: {
+    user: UserModel;
+  };
+};
+
+export default class CourseProgressListComponent extends Component<Signature> {
+  get courseParticipationGroups() {
+    const groupedCourseParticipations: CourseParticipationModel[][] = Object.values(groupBy(this.args.user.courseParticipations, 'course'));
+
+    const sortedParticipationsGroupedByCourse = groupedCourseParticipations.map((group) =>
+      group.sort((a, b) => new Date(a.lastSubmissionAt).getTime() - new Date(b.lastSubmissionAt).getTime()),
```

Comment:
Is `a.lastSubmissionAt` guaranteed to be not null?

Reply:
Okay looks like all the records we have at the moment have this as non-null. Let's add a database migration in core so that this is codified (and we never create invalid records in the future)